[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support

kadeng opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115877
* #115814
* #115813
* #115655
* #115654
* #112861
* #114687
* #115270
* #115174
* #115072
* #114606
* #114319
* #114075
* #113932
* #113891
* #113959
* __->__ #113890
* #113570
* #113569
* #113558
* #113399
* #113366
* #113365

This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

kadeng committed:
[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support

This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

aakhundov commented on torch/_inductor/kernel/bmm.py:
```diff
@@ -99,30 +104,50 @@ def tuned_bmm(mat1, mat2, *, layout=None):
 
 
 # Don't register this since it is slower than decomposing it
 # @register_lowering(aten.baddbmm)
```
Seems that this lowering is not registered?

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor cutlass backend] Enable torch.bmm and torch.baddbmm support"


This adds support for torch.bmm and torch.baddbmm operations through Cutlass backend. A comparatively simple addition.

[ghstack-poisoned]

kadeng commented:
Moved to a (draft) feature branch, see https://github.com/pytorch/pytorch/pull/115919
